### PR TITLE
Deferred static files

### PR DESCRIPTION
**Pull request by Archmonger**
This PR adds a `static_css` component, which reads the CSS file's from disk, stores the contents in cache (for performance purposes), and then puts the file contents into a `html.style` tag.

Currently, IDOM has no real way to "wait for CSS to load". This will lead to visual jank when attempting to use `html.link` to defer load CSS loading until a certain component exists. For example, visual jank occurs when using

```python
html.link({"rel": "stylesheet", "href": static("conreq/buttons.css")})
```

This also mitigates the need for the following (within Django only):
- https://github.com/idom-team/idom/issues/725

## Changelog
- Deferred static file components
- Docs for deferred static files
- Bump minimum IDOM version
- Create PR templates
- Sync issue form template with `idom-team/idom`
- Pin selenium to 4.2 for compatibility
- Bump django-idom to v1.1.0
- Docstring for component template tag
- Minor readme wordsmithing

### CODE REVIEW

File: src/django_idom/components.py
Comment (rmorshea) [inline src/django_idom/components.py:29]: If it weren't for the fact that IDOM is currently using `is` for string and float comparisons you could do this pretty cleanly with `use_memo`:

```python
from pathlib import Path

def static_css(static_path: str):
    abs_path = find(static_path)
    if not abs_path:
        raise FileNotFoundError(
            f"Could not find static file {static_path} within Django's static files."
        )

    return html.style(
        use_memo(
            lambda: Path(abs_path).read_text(encoding="utf-8"),
            dependencies=[abs_path, os.stat(abs_path).st_mtime],
        )
    )
```

I've already been thinking about special casing comparisons for ints, floats, strings etc to use `==` instead of `is` (which is basically how the `===` operator works in JS from what I can tell) and this example makes a good case for making that change.

Comment (rmorshea) [reply src/django_idom/components.py:29]: https://github.com/idom-team/idom/issues/773

Comment (Archmonger) [reply src/django_idom/components.py:29]: @rmorshea `use_memo` is not multiprocessing compatible, while `IDOM_CACHE` is.

**Review by rmorshea — APPROVED** [review]

**Comment by Archmonger** [comment]
@rmorshea This PR is still a draft, I have a bit more to work on.

I will mark as "Ready for review" when complete.

File: src/django_idom/components.py
Comment (Archmonger) [inline src/django_idom/components.py:10]: This brings up a bit of a debate.

I mentioned in the past that I prefer if we follow PEP8 styling for component names instead of React styling. This is mostly to avoid pissing off users that use Python style linters.

Whatever we decide on this will likely decide what we do in the future for any prefab components.

**Comment by Archmonger** [comment]
@rmorshea Would you have time to review this weekend?

**Comment by rmorshea** [comment]
I'll try to get to this today or tomorrow.

File: src/django_idom/components.py
Comment (Archmonger) [reply src/django_idom/components.py]: Going to assume we're pushing forward with PascalCase so I'm going to mark this as resolved.

**Comment by rmorshea** [comment]
apologies. i will make time tomorrow

File: src/django_idom/components.py
Comment (Archmonger) [reply src/django_idom/components.py]: While trying to convert Conreq component names to PascalCase, the pylint warnings started to bother me so I re-opened this.

File: .github/ISSUE_TEMPLATE/issue-form.yml
Comment (rmorshea) [inline .github/ISSUE_TEMPLATE/issue-form.yml:16]:
```suggestion
    required: false

```

File: src/django_idom/components.py
Comment (rmorshea) [inline src/django_idom/components.py:42]: Could you add a comment about the fact that this is preferable to `use_memo` because the cache works across-processes?

File: tests/test_app/static/static-css-test.css
Comment (rmorshea) [inline tests/test_app/static/static-css-test.css:3]:
```suggestion
}

```

File: tests/test_app/static/static-js-test.js
Comment (rmorshea) [inline tests/test_app/static/static-js-test.js:3]:
```suggestion
el.dataset.success = "true";

```

**Review by rmorshea — CHANGES_REQUESTED** [review]
Just a few minor things here and there. Besides that, LGTM!

File: src/django_idom/components.py
Comment (Archmonger) [reply src/django_idom/components.py]: @rmorshea Let's discuss this before PR merge.

This is important to hash out now since this is our first production-capable prefab component.

Comment (rmorshea) [reply src/django_idom/components.py]: I think for the sake of (1) alone, we should probably use snake case going forward.

Comment (rmorshea) [reply src/django_idom/components.py]: I also realized that making them camel case makes it seem like component functions can be used in type annotations, which they cannot.

**Comment by Archmonger** [comment]
Can't merge without re-approval.

**Review by rmorshea — APPROVED** [review]

**Pull request merged**
